Purchases/Networking/Caching/CallbackCache.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  CallbackCache.swift
+//
+//  Created by Joshua Liebowitz on 11/18/21.
+
+class CallbackCache<T> where T: CachableCallback {
```

Comment:
This is to reduce the copying of caching logic that we have for our requests.

Reply:
I love this so much

Reply:
ditto

Reply:
😁